agent/structs/config_entry.go

```diff
@@ -253,6 +320,30 @@ func (c *UpstreamConfiguration) Clone() *UpstreamConfiguration {
 	return &c2
 }
 
+// EndpointConfig represents a virtual service, i.e. one that is external to Consul
+type EndpointConfig struct {
```

Comment:
I left out protocol since it's already part of `service-defaults` and I think it's confusing to have it in two places.

I left in the TLS stuff because I think the spirit of this is that the tls authorization should be at the service level and not the gateway level.

Reply:
I think the TLS stuff should probably still continue to live in the gateway config entry.

These reference files that must already exist on the machine. `service-defaults` sorta represents the service in the abstract and may be used in a variety of consuming services on many machines whereas the terminating gateway config entry largely exists 1:1 with some actual real instance of the gateway running on a machine or small set of machines where you can arrange for those files to exist.